pulp_python/tests/functional/__init__.py

```diff
@@ -0,0 +1,88 @@
+try:
+    import pytest
+
+    from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+    from pulp_python.tests.functional.utils import gen_python_remote as gen_remote
+
+    from pulpcore.client.pulp_python import (
+        ApiClient,
+        ContentPackagesApi,
+        DistributionsPypiApi,
+        PublicationsPypiApi,
+        RepositoriesPythonApi,
+        RemotesPythonApi,
+    )
+except ImportError:
+    pass
+else:
+
+    # Bindings API Fixtures
+
+    @pytest.fixture
+    def python_bindings_client(cid, bindings_cfg):
+        """Provides the python bindings client object."""
+        api_client = ApiClient(bindings_cfg)
+        api_client.default_headers["Correlation-ID"] = cid
+        return api_client
+
+    @pytest.fixture
+    def python_repo_api_client(python_bindings_client):
+        """Provides the Python Repository API client object."""
+        return RepositoriesPythonApi(python_bindings_client)
+
+    @pytest.fixture
+    def python_distro_api_client(python_bindings_client):
+        """Provides the Python Distribution API client object."""
+        return DistributionsPypiApi(python_bindings_client)
+
+    @pytest.fixture
+    def python_content_api_client(python_bindings_client):
+        """Provides the Python Package Content API client object."""
+        return ContentPackagesApi(python_bindings_client)
+
+    @pytest.fixture
+    def python_remote_api_client(python_bindings_client):
+        """Provides the Python Remotes API client object."""
+        return RemotesPythonApi(python_bindings_client)
+
+    @pytest.fixture
+    def python_publication_api_client(python_bindings_client):
+        """Proves the Python Publication API client object."""
+        return PublicationsPypiApi(python_bindings_client)
+
+    # Object Generation Fixtures
+
+    @pytest.fixture
+    def python_repo(python_repo_api_client, gen_object_with_cleanup):
+        """Creates a Python Repository and deletes it at test cleanup time."""
+        return gen_object_with_cleanup(python_repo_api_client, gen_repo())
+
+    @pytest.fixture
+    def gen_python_distribution(python_distro_api_client, gen_object_with_cleanup):
```

Comment:
This is the current naming convention. Feel free to continue on with it. I believe we should switch at some point (it could be this PR!) to instead be `python_distribution_factory` which I believe is the naming convention pytest typically performs.